src/install/install.zig

```diff
@@ -10089,162 +10104,466 @@ pub const PackageManager = struct {
+                        ).asErr()) |e| {
+                            Output.prettyError("<r><red>error<r>: copying file {}", .{e});
+                            Global.crash();
+                        }
+
+                        var in_file = openFile(entry.dir, basename, .{ .mode = .read_only }) catch |e| {
+                            Output.prettyError("<r><red>error<r>: opening node_modules file {s} {s} ()", .{ @errorName(e), entrypath });
+                            Global.crash();
+                        };
+                        var in_file_close = true;
+                        defer if (in_file_close) in_file.close();
+
+                        const mode = in_file.mode() catch |e| {
+                            Output.prettyError("<r><red>error<r>: failed to get file mode {s} ", .{@errorName(e)});
+                            Global.crash();
+                        };
+                        var outfile = createFile(destination_dir_, entrypath, .{ .mode = mode }) catch |e| {
+                            Output.prettyError("<r><red>error<r>: failed to create file {s} ({s})", .{ entrypath, @errorName(e) });
+                            Global.crash();
+                        };
+                        var out_file_close = true;
+                        defer if (out_file_close) outfile.close();
+
+                        const infile_path = bun.path.joinStringBufWZ(buf1, &[_][]const u16{ in_dir, entry.path }, .auto);
+                        const outfile_path = bun.path.joinStringBufWZ(buf2, &[_][]const u16{ out_dir, entry.path }, .auto);
+
+                        in_file_close = false;
+                        out_file_close = false;
+                        in_file.close();
+                        outfile.close();
+                        bun.copyFileWithState(infile_path, outfile_path, &copy_file_state) catch |err| {
+                            Output.prettyError("<r><red>{s}<r>: copying file {}", .{ @errorName(err), bun.fmt.fmtOSPath(entry.path, .{}) });
+                            Global.crash();
+                        };
+
+                        continue;
+                    }
+
+                    var in_file = try openFile(entry.dir, entry.basename, .{ .mode = .read_only });
+                    defer in_file.close();
+
+                    @memcpy(pathbuf[0..entry.path.len], entry.path);
+                    pathbuf[entry.path.len] = 0;
+
+                    if (bun.sys.unlinkat(
+                        bun.toFD(destination_dir_.fd),
+                        pathbuf[0..entry.path.len :0],
+                    ).asErr()) |e| {
+                        Output.prettyError("<r><red>error<r>: copying file {}", .{e.withPath(entry.path)});
+                        Global.crash();
+                    }
+
+                    var outfile = try createFile(destination_dir_, entry.path, .{});
+                    defer outfile.close();
+
+                    if (comptime Environment.isPosix) {
```

Comment:
This code path is posix only, lets remove the `if` to avoid confusing the reader